src/Tribe/Attendee_Registration/Template.php

```diff
@@ -0,0 +1,325 @@
+
+	/**
+	 * Add the theme to the body class, in order to
+	 * add compatibility for official themes.
+	 *
+	 * @since TBD
+	 * @param array $classes List of classes to filter
+	 *
+	 * @return array $classes
+	 */
+	public function theme_body_class( $classes ) {
+
+		$child_theme  = get_option( 'stylesheet' );
+		$parent_theme = get_option( 'template' );
+
+		// if the 2 options are the same, then there is no child theme
+		if ( $child_theme == $parent_theme ) {
+			$child_theme = false;
+		}
+
+		if ( $child_theme ) {
+			$theme_classes = "tribe-theme-parent-$parent_theme tribe-theme-child-$child_theme";
+		} else {
+			$theme_classes = "tribe-theme-$parent_theme";
+		}
+
+		$classes[] = $theme_classes;
+
+		return $classes;
+	}
+
+
+	/**
+	 * Checks if theme needs a compatibility fix
+	 *
+	 * @since TBD
+	 * @param string $theme Name of template from WP_Theme->Template, defaults to current active template
+	 *
+	 * @return mixed
+	 */
+	public function theme_has_compatibility_fix( $theme = null ) {
+		// Defaults to current active theme
+		if ( null === $theme ) {
+			$theme = get_stylesheet();
+		}
+
+		// Return if the current theme is part of the ones we've compatibility for
+		return in_array( $theme, $this->themes_with_compatibility_fixes );
+	}
+
+	/**
+	 * This is where the magic happens where we run some ninja code that hooks
+	 * the query to resolve to an events template.
+	 *
+	 * @since TBD
+	 * @param WP_Query $query
+	 */
+	public function set_page_content( $query ) {
+
+		// Bail if we're not on the attendee info page
+		if ( ! tribe( 'tickets.attendee_registration' )->is_on_page() ) {
+			return;
+		}
+
+		if ( $this->is_main_loop( $query ) ) {
+			// on the_content, load our attendee info page
+			add_filter( 'the_content', array( tribe( 'tickets.attendee_registration.view' ), 'display_attendee_registration_page' ) );
+		}
+	}
+
+	/**
+	 * Modify the document title
+	 *
+	 * @since TBD
+	 * @param string $title
+	 *
+	 * @return array
+	 */
+	public function modify_page_title( $title ) {
+
+		// When in the loop, no need to override titles.
+		if ( in_the_loop() ) {
+			return $title;
+		}
+
+		if ( tribe( 'tickets.attendee_registration' )->is_on_page() ) {
+			$title['title'] = $this->get_page_title();
+		}
+
+		// Return the title
+		return $title;
+	}
+
+	/**
+	 * Return the Attendee Registration page title
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_page_title() {
+		/**
+		 * `tribe_tickets_attendee_registration_page_title`
+		 * Filters the attendee registration page title
+		 *
+		 * @param array $post_types Array of post types
+		 */
+		return apply_filters( 'tribe_tickets_attendee_registration_page_title', esc_html__( 'Attendee Registration', 'event-tickets' ) );
+	}
+
+	/**
+	 * Create a fake page we'll use to hijack our attendee info page
+	 *
+	 * @since TBD
+	 *
+	 * @return obj
+	 */
+	public function spoofed_page() {
+
+		$spoofed_page = array(
+			'ID'                    => -1,
```

Comment:
Has this been tested with translation plugins? I know many do not appreciate non-valid post IDs and will hide the page as not a translated one.

Reply:
Good catch! No, we use something similar for the events page on TEC. Ideally, this template will only parse the elements that are on the cart and don't have the meta information completed (all required fields). I think then, the job to make it compatible with translation plugins would be more related to the slug. Just thinking out loud, haven't thought deep about it.